geopaparazzispatialitelibrary/src/eu/geopaparazzi/spatialite/database/spatial/core/daos/SPL_Rasterlite.java

```diff
@@ -187,19 +190,289 @@ public static boolean hasRasterLiteSupport(){
+        qSb.append(",'");
+        qSb.append(styleNameVector);
+        qSb.append("','");
+        qSb.append(mimeType);
+        qSb.append("','");
+        qSb.append(bgColor);
+        qSb.append("',");
+        qSb.append(Integer.toString(transparent));
+        qSb.append(",");
+        qSb.append(Integer.toString(quality));
+        qSb.append(",");
+        qSb.append(Integer.toString(reaspect));
+        qSb.append(");");
+        String s_sql_command = qSb.toString();
+        Stmt stmt = null;
+        byte[] ba_image = null;
+        if (!SPL_Rasterlite.Rasterlite2Version_CPU.equals("")) { // only if rasterlite2 driver is active
+            try {
+                stmt = dbSpatialite.prepare(s_sql_command);
+                if (stmt.step()) {
+                    ba_image = stmt.column_bytes(0);
+                }
+            } catch (jsqlite.Exception e_stmt) {
+                /*
+                  this internal lib error is not being caught and the application crashes
+                  - the request was for a image 1/3 of the orignal size of 10607x8292 (3535x2764)
+                  - big images should be avoided, since the application dies
+                  'libc    : Fatal signal 11 (SIGSEGV) at 0x80c7a000 (code=1), thread 4216 (AsyncTask #2)'
+                  '/data/app-lib/eu.hydrologis.geopaparazzi-2/libjsqlite.so (rl2_raster_decode+8248)'
+                  'I WindowState: WIN DEATH: Window{41ee0100 u0 eu.hydrologis.geopaparazzi/eu.hydrologis.geopaparazzi.GeoPaparazziActivity}'
+                */
+                int i_rc = dbSpatialite.last_error();
+                GPLog.error("SPL_Rasterlite", "rl2_GetMapImageFromVector sql[" + s_sql_command + "] rc=" + i_rc + "]", e_stmt);
+            } finally {
+                if(stmt!=null)
+                    try {
+                        stmt.close();
+                    } catch (jsqlite.Exception e) {
+                        GPLog.error("SPL_Rasterlite", "rl2_GetMapImageFromVector sql[" + s_sql_command + "]", e);
                     }
             }
         }
         return ba_image;
     }
+    /**
+     *  HashMap<Integer, Double> of Zoom-Levels 0-30
+     *  Zoom-Level, Width in World-Mercator-Meters of a 256x256 Tile
+     *  - 'WGS 84 / World Mercator' (3395) - UNIT=Meters            
+     * @return zoom_levels
+     */     
+    public static int[] rl2_calculate_zoom_levels(Database dbSpatialite, String coverageName, String s_tile_size,int i_srid) {
+        HashMap<Integer, Double> zoom_levels = new HashMap<Integer, Double>();  
+        // TODO: remove to Mercator-specific definitions
+        // 'WGS 84 / World Mercator' (3395) - UNIT=Meters            
+        // Zoom-Level, Width in Meters of a 256x256 Tile
+        // Value returned by gdalinfo for a geo-referenced Tile
+        // values based on the position: 13.37771496361961 52.51628011262304 [Brandenburg Gate, Berlin, Germany]
+        zoom_levels.put( 0,40032406.294);
```

Comment:
Ok, it might be a good idea to put the zoomlevels in a central place. It seems like a reusable information to me. I will do that after the merge of the pull request.